#!/usr/bin/env python
PACKAGE = "fotonic_e70"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()



fps = gen.enum([ gen.const("frame_rate_30",  int_t, 30, ""),
                 gen.const("frame_rate_60",  int_t, 60, "")],
                 "frame rate")

gen.add("fps", int_t, 0, "Frames per second", 30, edit_method=fps)
gen.add("shutter_speed", double_t, 0, "Shutterspeed [ms]", 200.0, 20.0, 500.0)
gen.add("fps_divisor", int_t, 0, "FPS divisor", 1, 1, 10)
gen.add("edge_filter", bool_t, 0, "Filter edges", True)


exit(gen.generate(PACKAGE, "fotonic_e70", "FotonicE70"))
